lnrpc/routerrpc/router_backend.go

```diff
@@ -236,3 +255,165 @@ func (r *RouterBackend) MarshallRoute(route *routing.Route) *lnrpc.Route {
 
 	return resp
 }
+
+// ExtractIntentFromSendRequest attempts to parse the SendRequest details
+// required to dispatch a client from the information presented by an RPC
+// client.
+func (r *RouterBackend) ExtractIntentFromSendRequest(rpcPayReq *PaymentRequest) (
```

Comment:
Seems to duplicate an existing method in the RPC server without removing the existing one. Also if this is to be done, then I think it can be done in a distinct commit.

Reply:
I had a common method before, but then we introduced some differences that made not easily possible to reuse. The rpc request message is different. Fee limit is expressed differently and a custom timeout field is added.